src/main/scala/io/citrine/lolo/trees/multitask/MultiTaskTrainingNode.scala

```diff
@@ -2,171 +2,207 @@ package io.citrine.lolo.trees.multitask
 
 import io.citrine.lolo.PredictionResult
 import io.citrine.lolo.linear.GuessTheMeanLearner
+import io.citrine.lolo.trees.classification.ClassificationTrainingLeaf
 import io.citrine.lolo.trees.regression.RegressionTrainingLeaf
-import io.citrine.lolo.trees.splits.{MultiTaskSplitter, NoSplit}
-import io.citrine.lolo.trees.{InternalModelNode, ModelNode, TrainingLeaf}
+import io.citrine.lolo.trees.splits.{MultiTaskSplitter, NoSplit, Split}
+import io.citrine.lolo.trees.{InternalModelNode, ModelNode, TrainingNode}
 import io.citrine.random.Random
 
 import scala.collection.mutable
 
-/**
-  * Node in a multi-task training tree, which can produce nodes for its model trees.
-  * Splits are chosen using a MultiTaskSplitter, which considers the sum impurity decrease across all tasks.
-  *
-  * @param inputs data on which to select splits and form models
-  * @param numFeatures to randomly select from at each split (negative values indicate that all features should be considered)
-  * @param maxDepth to grow the tree to
-  * @param minInstances minimum number of training instances per leaf
-  * @param randomizePivotLocation whether to generate splits randomly between the data points
-  * @param rng random number generator, for reproducibility
-  */
-class MultiTaskTrainingNode(
-    inputs: Seq[(Vector[AnyVal], Array[AnyVal], Double)],
-    numFeatures: Int,
-    maxDepth: Int,
-    minInstances: Int,
-    randomizePivotLocation: Boolean = false,
-    rng: Random
+case class MultiTaskTrainingNode(
+    trainingData: Seq[(Vector[AnyVal], Array[AnyVal], Double)],
```

Comment:
Why are we using an `Array[AnyVal]` for the labels type? Seems odd to be passing around an immutable `Vector` on one end and the mutable collection on the other if we're not using that aspect of the collection.

Reply:
I don't know; that's historical. I'll change it